code/_helpers/detection.dm

```diff
@@ -58,31 +58,35 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 	return view(range, GLOB.dview_mob)
 
 //Returns a list of all turfs this mob can see, accounting for view radius and offset
-/atom/proc/turfs_in_view(var/check_range = world.view)
```

Comment:
The intended purpose of dview is to see in the dark, ignoring byond-native darkness that obscures things. Because it is an unreliable and largely obsolete interference we don't use

Also a secondary consideration, the possibility that the atom is inside something, we generally want this proc to return a view from the toplevel.

Lastly though, i think the conclusion of concurrency issues is questionable. DVIEW operates in a single frame as far as i'm aware, i don't see any possibility of interference even if a million things are all trying to use it

Reply:
The initial proc just set sight_flags to 0 every time rather than utilising the sight flags. A view from the top-level is a valid argument, I can account for this.
RE: Concurrency. I still think it's a possible explanation for this problem. That dview mob was intended for lighting (citation needed), not to be accessed by a load of mobs simultaneously. The START_DVIEW macro makes no check / consideration for whether the dview mob is actually available, and it's entirely possible that two concurrent calls to it could cause this weirdness with dir. It may well operate in a single frame, but what about the callers? under load, is it not feasible that two mobs could have their calls to it queued up and executed at almost exactly the same time.